dot/network/config.go

```diff
@@ -10,19 +10,20 @@ import (
 	"path"
 	"time"
 
+	"github.com/adrg/xdg"
 	"github.com/libp2p/go-libp2p/core/crypto"
 
 	"github.com/ChainSafe/gossamer/internal/log"
 	"github.com/ChainSafe/gossamer/internal/metrics"
 	"github.com/ChainSafe/gossamer/lib/common"
 )
 
-const (
+var (
 	// DefaultKeyFile the default value for KeyFile
 	DefaultKeyFile = "node.key"
 
 	// DefaultBasePath the default value for Config.BasePath
-	DefaultBasePath = "~/.gossamer/gssmr"
+	DefaultBasePath = xdg.DataHome + "/gossamer"
```

Comment:
We can change `DefaultBasePath` as var since xdg.DataHome is a var. But we can keep other values as const.